#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_TuxFamily-suphp-0.7.1-3-add-targetmode-execute.dpatch  
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: TuxFamily patch to add support for executing scripts with a chosen interpreter

@DPATCH@
diff -Nru a/src/Application.cpp b/src/Application.cpp
--- a/src/Application.cpp	2011-04-17 11:27:28.000000000 +0000
+++ b/src/Application.cpp	2011-04-17 11:30:19.000000000 +0000
@@ -583,6 +583,8 @@
         return TARGETMODE_PHP;
     else if (interpreter == "execute:!self")
         return TARGETMODE_SELFEXECUTE;
+    else if (interpreter.substr(0, 8) == "execute:")
+        return TARGETMODE_EXECUTE;
     else
         throw SecurityException("Unknown Interpreter: " + interpreter,
                                 __FILE__, __LINE__);
@@ -608,6 +610,12 @@
             CommandLine cline;
             cline.putArgument(scriptFilename);
             API_Helper::getSystemAPI().execute(scriptFilename, cline, env);
+        } else if (mode == TARGETMODE_EXECUTE) {
+            std::string interpreterPath = interpreter.substr(8);
+            CommandLine cline;
+            cline.putArgument(interpreterPath);
+            cline.putArgument(scriptFilename);
+            API_Helper::getSystemAPI().execute(interpreterPath, cline, env);
         }
     } catch (SystemException& e) {
         throw SoftException("Could not execute script \"" + scriptFilename
diff -Nru a/src/Application.hpp b/src/Application.hpp
--- a/src/Application.hpp	2011-04-17 10:15:36.000000000 +0000
+++ b/src/Application.hpp	2011-04-17 11:27:40.000000000 +0000
@@ -26,6 +26,7 @@
 
 enum TargetMode {
     TARGETMODE_PHP,
+    TARGETMODE_EXECUTE,
     TARGETMODE_SELFEXECUTE
 };
 
